range_queries/fenwick_tree.cpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @brief Fenwick tree
+ *
+ * A Fenwick tree or binary indexed tree is a data structure 
+ * that can efficiently update elements and calculate 
+ * prefix sums in a table of numbers.
+ */
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+class FenwickTree {
```

Comment:
Could you document the classes and members as shown in there cue above

Reply:
I documented some. Please take a look.